packages/firestore/test/unit/specs/spec_test_runner.ts

```diff
@@ -1190,7 +1190,10 @@ export interface SpecWatchEntity {
  * Note that the last parameter is really of type ...string (spread operator)
  * The filter is based of a list of keys to match in the existence filter
  */
-export type SpecWatchFilter = [TargetId[], string];
+export interface SpecWatchFilter extends Array<TargetId[] | string> {
+  '0': TargetId[];
```

Comment:
This is surprising that this works. Thanks for digging into it.

Reply:
I found it when I was testing stricter tuples for Typescript 2.7. The new definition is basically equivalent to the old meaning of 'tuple' in Typescript.